internal/rename/namespace/executor.go

```diff
@@ -0,0 +1,41 @@
+/*
+ *  *******************************************************************************
+ *  * Copyright (c) 2019 Edgeworx, Inc.
+ *  *
+ *  * This program and the accompanying materials are made available under the
+ *  * terms of the Eclipse Public License v. 2.0 which is available at
+ *  * http://www.eclipse.org/legal/epl-2.0
+ *  *
+ *  * SPDX-License-Identifier: EPL-2.0
+ *  *******************************************************************************
+ *
+ */
+
+package namespace
+
+import (
+	"github.com/eclipse-iofog/iofogctl/internal/config"
+	"github.com/eclipse-iofog/iofogctl/pkg/util"
+)
+
+func Execute( namespace, name, newname string) error {
+	namespace, err := config.GetNamespace(exe.name)
+	if err != nil {
+		return err
+	}
+	if name == "" || name == "default" {
+		return util.NewError("Cannot rename default or nonexistant namespaces")
+	}
+
+	// Check for namespace
+	for i, n := range conf.Namespaces {
+		if n == name {
+			conf.Namespaces[i].name = newname
+			util.PrintSuccess("Renamed namespace " + name + "to " + newname)
+			return nil
```

Comment:
You need to flush, but the logic is more complex, as you will need to update ~/.iofog/config.yaml, and rename the corresponding file ~/.iofog/namespaces/<NAME>.yaml